cmd/root.go

```diff
@@ -77,6 +78,17 @@ func runRootCmd(cmd *cobra.Command, args []string) {
 			if err = viper.ReadInConfig(); err != nil {
 				logger.Fatalw("failed to read config", zap.Error(err))
 			}
+
+			var inputs []interface{}
+			for _, v := range inputsource.All() {
+				inputs = append(inputs, v)
+			}
+			viper.Set("input_sources.primary", inputs)
```

Comment:
probably `viper.Set("input_sources.primary", inputsource.All())` also works?

I don't get why do we have an extra loop at lines 82-85

Reply:
That's not going to work. `viper.Set` accepts `[]interface{}` while `inputsource.All()` returns `[]string`.